generator/mavgen_java.py

```diff
@@ -60,48 +58,57 @@ def generate_CRC(directory, xml):
 
 package com.MAVLink.${basename};
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
-* X.25 CRC calculation for MAVlink messages. The checksum must be initialized,
-* updated with witch field of the message, and then finished with the message
-* id.
-*
-*/
+ * X.25 CRC calculation for MAVlink messages. The checksum must be initialized,
+ * updated with witch field of the message, and then finished with the message
+ * id.
+ */
 public class CRC {
-    private static final int[] MAVLINK_MESSAGE_CRCS = {${message_crcs_array}};
+    private static final Map<Integer, Integer> MAVLINK_MESSAGE_CRCS;
     private static final int CRC_INIT_VALUE = 0xffff;
     private int crcValue;
 
+    static {
+        MAVLINK_MESSAGE_CRCS = new HashMap<>();
+        ${message_crcs_array}
+    }
+
     /**
-    * Accumulate the X.25 CRC by adding one char at a time.
-    *
-    * The checksum function adds the hash of one char at a time to the 16 bit
-    * checksum (uint16_t).
-    *
-    * @param data
-    *            new char to hash
-    **/
-    public  void update_checksum(int data) {
+     * Accumulate the X.25 CRC by adding one char at a time.
+     *
+     * The checksum function adds the hash of one char at a time to the 16 bit
+     * checksum (uint16_t).
+     *
+     * @param data new char to hash
+     **/
+    public void update_checksum(int data) {
         data = data & 0xff; //cast because we want an unsigned type
         int tmp = data ^ (crcValue & 0xff);
         tmp ^= (tmp << 4) & 0xff;
         crcValue = ((crcValue >> 8) & 0xff) ^ (tmp << 8) ^ (tmp << 3) ^ ((tmp >> 4) & 0xf);
     }
 
     /**
-    * Finish the CRC calculation of a message, by running the CRC with the
-    * Magic Byte. This Magic byte has been defined in MAVlink v1.0.
-    *
-    * @param msgid
-    *            The message id number
-    */
-    public void finish_checksum(int msgid) {
-        update_checksum(MAVLINK_MESSAGE_CRCS[msgid]);
+     * Finish the CRC calculation of a message, by running the CRC with the
+     * Magic Byte.
+     *
+     * @param msgid The message id number
+     * @return boolean True if the checksum was successfully finished. Otherwise false
+     */
+    public boolean finish_checksum(int msgid) {
+        if (MAVLINK_MESSAGE_CRCS.containsKey(msgid)) {
```

Comment:
@WickedShell My solution to our [discussion](https://github.com/ArduPilot/pymavlink/pull/341#discussion_r305652662) in the previous PR is to return a boolean back up to the Parser. If CRC extra does not exist, a false is passed back up and the current packet is tossed out